src/main/java/com/flightstats/hub/dao/aws/HubS3Client.java

```diff
@@ -108,6 +108,15 @@ void deleteObject(DeleteObjectRequest request) {
         }
     }
 
+    void deleteVersion(DeleteVersionRequest request) {
+        try {
+            s3Client.deleteVersion(request);
+        } catch (SdkClientException e) {
+            countError(e, request, "deleteVersion", Arrays.asList("bucket:" + request.getBucketName(), "key:" + request.getKey()));
+            throw e;
+        }
+    }
```

Comment:
Why is there a manual delete on s3 object versions? Isn't this managed thru s3 bucket lifecycle policies?

Reply:
It could be handled through lifecycle policy, but as far as I can tell that would use another of our limited LPs, by immediately deleting the version manually I am trying to avoid any costs of orphaned versions while leaving the LPs to be available for consumer channel TTLs

Reply:
We know that Lifecycle policies offered by S3 bucket is not sufficient to cleanup stale data in S3 bucket. We do need an alternative/additional solution to accomplish this. IMO, it is not good idea to introduce another approach to cleaning up data.

Reply:
I feel like this is pretty isolated for an isolated problem, but I see your point